payment_mollie_official/models/mollie.py

```diff
@@ -129,39 +134,33 @@ def _mollie_form_validate(self, data):
 
         _logger.info('Validated transfer payment for tx %s: set as pending' % (reference))
         mollie_api_key = acquirer._get_mollie_api_keys(acquirer.environment)['mollie_api_key']
-        url = "%s/payments" % (acquirer._get_mollie_urls(acquirer.environment)['mollie_form_url'])
+        url_base = acquirer._get_mollie_urls(acquirer.environment)['mollie_form_url']
+        url_payments = urljoin(url_base, "payments")
+        url = urljoin(url_payments, transactionId)
 
-        payload = {
-            "id": transactionId
-        }
         if acquirer.environment == 'test':
-            payload["testmode"] = True
+            payload = {"testmode": True}
+        else:
+            payload = {}
 
         headers = {'content-type': 'application/json', 'Authorization': 'Bearer ' + mollie_api_key}
 
+        _logger.info("GET ", url)
         mollie_response = requests.get(
-            url, data=json.dumps(payload), headers=headers).json()
+            url, params=payload, headers=headers).json()
 
         if self.state == 'done':
             _logger.info('Mollie: trying to validate an already validated tx (ref %s)', reference)
             return True
 
-        data_list = mollie_response["data"]
-        data = {}
-        status = 'undefined'
-        for x in data_list:
-            if isinstance(x, dict) and x.get('id', False) and x['id'] == \
-                    transactionId:
-                data = x
-                break
-
-        if "status" in data:
-            status = data["status"]
+        status = mollie_response.get("status", "undefined")
```

Comment:
We may need to check that the result returned by mollie does match the transaction ID we sended. Like it's done on line 153 (at the end). Else we should raise an error and report it in the log.

Reply:
Resolved.